src/OSPSuite.Core/Services/GarbageCollectionTask.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using System.Runtime;
+
+namespace OSPSuite.Core.Services
+{
+   public static class GarbageCollectionTask
+   {
+      /// <summary>
+      ///    Force GC Collection compacting the Large Object Heap. Implementation very similar to what DotMemory
+      ///    is doing according to one of their post
+      /// </summary>
+      public static void ForceGC()
+      {
+         GCSettings.LargeObjectHeapCompactionMode = GCLargeObjectHeapCompactionMode.CompactOnce;
+         for (int i = 0; i < 4; i++)
+         {
+            GC.Collect(2, GCCollectionMode.Forced, blocking: true);
+            GC.WaitForPendingFinalizers();
+         }
```

Comment:
where does the 4 in the GC-loop actually come from?

Reply:
no idea. Empirical value apparently that mimics so behavior. There is always stuff left if you do it only 3 times or less

Reply:
hmm, ok :)

Reply:
I have added  a link to the discussion in code
https://dotnettools-support.jetbrains.com/hc/en-us/community/posts/360000088690-How-reproduce-DotMemory-s-Force-GC-button-s-behaviour-on-code-with-c-